src/scripts/main.js

```diff
@@ -1,3 +1,63 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', function () {
+  const appendRowBtn = document.querySelector('.append-row');
+  const removeRowBtn = document.querySelector('.remove-row');
+  const appendColumnBtn = document.querySelector('.append-column');
+  const removeColumnBtn = document.querySelector('.remove-column');
+  const table = document.querySelector('.field');
+
+  const maxRows = 10;
+  const minRows = 2;
+  const maxColumns = 10;
+  const minColumns = 2;
+
+  function updateButtonStates() {
+    const numRows = table.rows.length;
+    const numColumns = table.rows[0].cells.length;
+
+    appendRowBtn.disabled = numRows >= maxRows;
+    removeRowBtn.disabled = numRows <= minRows;
+    appendColumnBtn.disabled = numColumns >= maxColumns;
+    removeColumnBtn.disabled = numColumns <= minColumns;
+  }
+
+  appendRowBtn.addEventListener('click', function () {
```

Comment:
You can use the arrow function
```suggestion
  appendRowBtn.addEventListener('click', () => {
```